File: src/main/scala/com/chandy/spark/Application.scala
package com.chandy.spark

import org.apache.spark.{SparkConf, SparkContext}

/**
  * @Author: chandy1994
  * @Date: 2019/06/12 0:27
  */
object Application {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("test").setMaster("local[4]")
    val sc = new SparkContext(conf)
    val rdd1 = sc.parallelize(List(1, 2, 3))
    print(rdd1.reduce(_ + _))
  }

}

File: src/main/java/com/chandy/spark/app/Application.java
package com.chandy.spark.app;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;
import org.apache.spark.rdd.IntegerRDD;

/**
 * @Author: chandy1994
 * @Date: 2019/06/12 0:38
 */
public class Application {
    public static void main(String[] args) throws InterruptedException {
        SparkConf sparkConf = new SparkConf().setMaster("local[*]").setAppName("spark-custom-RDD");
        JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);

        JavaRDD<Integer> javaRDD1 = new IntegerRDD(sparkContext.sc()).toJavaRDD();
        JavaRDD<Integer> javaRDD2 = new IntegerRDD(sparkContext.sc()).toJavaRDD();
        javaRDD1.union(javaRDD2).distinct().foreach(new VoidFunction<Integer>() {
            public void call(Integer integer) throws Exception {
                System.out.println(integer);
            }
        });

        sparkContext.close();
    }
}
